src/extensions/CanvasWidget/CanvasWidget.tsx

```diff
@@ -209,25 +202,37 @@ const CanvasWidget = <T extends BaseLayerProps>(props: Props<T>) => {
     }, [])
 
     const _handleWheel = useCallback((e: React.WheelEvent<HTMLCanvasElement>) => {
-        if (!layers) return
         for (let l of layers) {
-            l.handleWheelEvent(e)
+            if (l) {
+                l.handleWheelEvent(e)
+            }
         }
     }, [layers])
 
     const _handleKeyPress = useCallback((e: React.KeyboardEvent<HTMLDivElement>) => {
-        if (!layers) return
         for (let l of layers) {
-            if (l.handleKeyboardEvent(KeyEventType.Press, e) === false) {
-                e.preventDefault()
+            if (l) {
+                if (l.handleKeyboardEvent(KeyEventType.Press, e) === false) {
+                    e.preventDefault()
+                }
             }
         }
     }, [layers])
 
+    // make sure that the layers have width/height that matches this canvas widget
+    for (const L of layers) {
+        if (L) {
+            const {width: layerWidth, height: layerHeight} = L.getProps()
+            if ((layerWidth !== width) || (layerHeight !== height)) {
```

Comment:
We check that the width/height sent to the canvaswidget is the same as those passed to each layer.